flow/Net2.actor.cpp

```diff
@@ -900,8 +901,12 @@ void ConfigureSSLContext( const LoadedTLSConfig& loaded, boost::asio::ssl::conte
 			Reference<TLSPolicy> tlsPolicy = Reference<TLSPolicy>(new TLSPolicy(loaded.getEndpointType()));
 			tlsPolicy->set_verify_peers({ loaded.getVerifyPeers() });
 
-			context->set_verify_callback([policy=tlsPolicy](bool preverified, boost::asio::ssl::verify_context& ctx) {
-						return policy->verify_peer(preverified, ctx.native_handle());
+			context->set_verify_callback([policy=tlsPolicy, onPolicyFailure](bool preverified, boost::asio::ssl::verify_context& ctx) {
+						bool success = policy->verify_peer(preverified, ctx.native_handle());
```

Comment:
Are there any client TLS misconfigurations that would result in failures elsewhere (e.g. in the handshake or something)? If so, is it feasible to capture them all?

Reply:
I think in the general sense, yes.  The client could present a mismatching set of encryption and signing algorithms, or disconnect during the handshake, or send garbage data.  Most of these will be highly unlikely as we control the one and only client, which is produced in tandem with the server.

There's other errors like the client could kill the handshake because the server failed the client's verify peers check, but I'm not sure how to differentiate that from a normal handshake close.

So I think the real answer is, as far as I know with my current knowledge of TLS, this covers most of what we would expect and have previously seen cause problems in production.